### PR TITLE
introducing project flavors

### PR DESCRIPTION
**Pull request by nopeslide**
## Related Issues & PRs

none

## Description

- [x] moved main project into lifesensor
- [x] renamed test project to lifesensor_test
- [x] made clean target work without docker or idf
- [x] added makefile target reconfigure
- [x] updated README.md
- [x] refactored .gitignores
- [x] updated github actions
- [x] refactored Dockerfiles, added missing deps
- [x] propagate DOCKEROPTS variable from major Makefile to docker Makefile
- [x] hid main Makefile and introduced flavor specific ones, which use the hidden main Makefile 

## Context

I moved the project into subdirs to allow for project flavors like unit_testing or coming simulation specific stuff.

## Testing

- [x] tested building
- [x] tested flashing
- [x] tested monitoring 

## Documentation

- [x] documented new Makefile variable

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,17 +4,27 @@ ESP32 firmware for the lifesensor project.
 ## Components
 See [components/README.md](components/README.md) for an overview of features. 
 
+## Projects flavors
```
Comment (nopeslide) [inline README.md]:
```suggestion
## Project flavors
```

File: README.md [hunk 2]
```diff
@@ -24,24 +34,8 @@ See [components/README.md](components/README.md) for an overview of features.
...
-  * Run `make help` to see all options
-  1. `make test`
-     1. creates idf docker image
-     2. starts idf docker container to build test project (`make build-test`)
-     3. starts idf docker container to flash the test project to ESP via specified device by `make dev` (`make flash-test`)
-     4. starts idf docker container to connect to ESP via specified device by `make dev` (`make monitor-test`)
-* via local esd-idf toolchain
-  1. Change to test project
-     * `cd test`
+  1. Choosing main project
```
Comment (nopeslide) [inline README.md]:
```suggestion
  1. Choosing project flavor
```

**Review by nopeslide — COMMENTED** [review]
fixed some typos

**Review by nopeslide — COMMENTED** [review]
think I got everything now

**Comment by nopeslide** [comment]
cleaned up PR

**Review by ikstream — CHANGES_REQUESTED** [review]
lgtm, but would be good to test it very extensively first, as with #56 
@pascalharp @jhonnyam @reporiot

**Review by reporiot — COMMENTED** [review]
lgtm

**Review by ikstream — APPROVED** [review]

**Pull request merged**